include/seqan3/alignment/pairwise/alignment_configurator.hpp

```diff
@@ -157,33 +165,199 @@ struct alignment_configurator
+
+    // ----------------------------------------------------------------------------
+    // affine gap kernel
+    // ----------------------------------------------------------------------------
+
+    using affine_t = deferred_crtp_base<affine_gap_policy, cell_type>;
+
+    // ----------------------------------------------------------------------------
+    // configure initialisation policy
+    // ----------------------------------------------------------------------------
+
+    // Get the value for the sequence ends configuration.
+    auto align_ends_cfg = cfg.template value_or<align_cfg::aligned_ends>(align_cfg::none_ends_free);
+    using align_ends_cfg_t = decltype(align_ends_cfg);
+
+    // This lambda augments the initialisation policy of the alignment algorithm
+    // with the aligned_ends configuration from before.
+    auto _leading_both = [&](auto first_seq, auto second_seq) constexpr
+    {
+        // Define the trait for the initialisation policy
+        struct _trait
+        {
+            using free_first_leading_t  [[maybe_unused]] = decltype(first_seq);
+            using free_second_leading_t [[maybe_unused]] = decltype(second_seq);
+        };
+
+        // Make initialisation policy a deferred CRTP base and delegate to configure the find optimum policy.
+        using init_t = deferred_crtp_base<affine_gap_init_policy, _trait>;
+        return configure_free_ends_optimum_search<kernel_t, affine_t, dp_matrix_t, init_t>(cfg);
+    };
+
+    // This lambda determines the initialisation configuration for the second sequence given
+    // the leading gap property for it.
+    auto _leading_second = [&](auto first) constexpr
+    {
+        // If possible use static information.
+        if constexpr (align_ends_cfg_t::template is_static<2>())
+        {
+            return _leading_both(first, std::integral_constant<bool, align_ends_cfg_t::template get_static<2>()>{});
+        }
+        else
+        {   // Resolve correct property at runtime.
+            if (align_ends_cfg[2])
+                return _leading_both(first, std::true_type{});
+            else
+                return _leading_both(first, std::false_type{});
+        }
+    };
+
+    // Here the initialisation configuration for the first sequence is determined given
+    // the leading gap property for it.
+    // If possible use static information.
+    if constexpr (align_ends_cfg_t::template is_static<0>())
+    {
+        return _leading_second(std::integral_constant<bool, align_ends_cfg_t::template get_static<0>()>{});
+    }
+    else
+    {  // Resolve correct property at runtime.
+        if (align_ends_cfg[0])
+            return _leading_second(std::true_type{});
+        else
+            return _leading_second(std::false_type{});
+    }
+}
+//!\endcond
+
+//!\cond
+template <typename kernel_t, typename ...policies_t, typename config_t>
+constexpr auto
+alignment_configurator::configure_free_ends_optimum_search(config_t const & cfg)
```

Comment:
one line

Reply:
set return type to std::function or do this in a comment (don't need to compress into one line then)